### PR TITLE
DAOS-14264 test: Include IntelMPI in soak testing

### PR DESCRIPTION
**Pull request by mjean308**
Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: soak_smoke

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/soak/soak-extra-intelmpi.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+vpic_smoke:
+  module: mpi/latest
```
Comment (daltonbohning) [inline src/tests/ftest/soak/soak-extra-intelmpi.yaml:2]: How does `mpi/latest` correspond to intel mpi? Is this Aurora specific?

File: src/tests/ftest/util/soak_utils.py [hunk 2]
```diff
@@ -1199,9 +1200,20 @@ def create_app_cmdline(self, job_spec, pool, ppn, nodesperjob):
     """
     commands = []
     app_params = os.path.join(os.sep, "run", job_spec, "*")
-    app_cmd = os.path.expandvars(self.params.get("cmdline", app_params, default=None))
     mpi_module = self.params.get("module", app_params, self.mpi_module)
     api_list = self.params.get("api", app_params, default=["DFS"])
+    apps_dir = os.environ["DAOS_TEST_APP_DIR"]
+    # Update DAOS_TEST_APP_DIR if used in the cmdline param in yaml
+    # ${DAOS_TEST_APP_SRC}                  =>  apps built with el8 and mpi/mpich (default)
+    # ${DAOS_TEST_APP_SRC}/intelmpi         =>  apps built with el8 and intelmpi
```
Comment (daltonbohning) [inline src/tests/ftest/util/soak_utils.py:1208]: > GitHub Actions / Pylint check
>
> wrong-spelling-in-comment, Wrong spelling of a word 'intelmpi' in a comment:

Either remove this line are you can put this above it
```
# pylint: disable-next=wrong-spelling-in-comment,fixme
```

(And two more instances below)

Comment (daltonbohning) [reply src/tests/ftest/util/soak_utils.py:1209]: Sorry, I didn't mean to copy-past the `fixme` part :(
It should be
`# pylint: disable-next=wrong-spelling-in-comment`

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by rpadma2 — APPROVED** [review]

**Pull request merged**